File: src/jimin/week3/삼각 달팽이.py
def solution(n):
    answer = []
    arr = list([0] * n for _ in range(n))
    y, x = -1, 0
    count = 1
    for i in range(n):
        for j in range(i, n):
            if i % 3 == 0:
                y += 1
            elif i % 3 == 1:
                x += 1
            else:
                x -= 1
                y -= 1
            arr[y][x] = count
            count += 1
    for i in range(n):
        for j in range(0, i + 1):
            answer.append(arr[i][j])

    return answer


n = 4
print(solution(n))

File: src/taeeun/week3/삼각 달팽이.js
function solution(n) {
    var answer = [];
    let arr = Array.from(Array(n), (el, i) => Array(i+1))
    let currentX=0;
    let currentY=0;
    const totalMove = n*(n+1)/2;
    let cnt=0;
    let change=n;
    
    //방향은 총 세가지 1: 아래, 2: 옆, 3: 위
    let dir=1;
    
    for (let i=1; i<=totalMove; i++){ 
        arr[currentX][currentY]=i;
        cnt++;
        //방향을 바꿔야하는지?
        if (cnt===change) {
            dir=dir%3===0 ? 1: dir+1;
            cnt=0;
            change-=1;
        }
        if (dir===1) {
            currentX+=1;
            continue;
        }
        if (dir===2) {
            currentY+=1;
            continue;
        }
        if (dir===3) {
            currentX-=1;
            currentY-=1;
        }
    }
    

    return arr.flat();
}
